kubernetes-ingress.el

```diff
@@ -0,0 +1,144 @@
+;;; kubernetes-ingress.el --- Rendering routines for Kubernetes ingress.  -*- lexical-binding: t; -*-
+;;; Commentary:
+;;; Code:
+
+(require 'dash)
+
+(require 'kubernetes-ast)
+(require 'kubernetes-loading-container)
+(require 'kubernetes-modes)
+(require 'kubernetes-props)
+(require 'kubernetes-state)
+(require 'kubernetes-utils)
+(require 'kubernetes-yaml)
+
+
+;; Components
+
+(defconst kubernetes-ingress--column-heading
+  (propertize (format "%-45s %-25s %20s %10s" "Name" "Hosts" "Address"  "Age")
+              'face 'magit-section-heading))
+
+(kubernetes-ast-define-component ingress-detail (ingress)
+  (-let [(&alist 'metadata (&alist 'namespace ns 'creationTimestamp time)) ingress]
+    `((section (namespace nil)
+               (nav-prop (:namespace-name ,ns)
+                         (key-value 12 "Namespace" ,(propertize ns 'face 'kubernetes-namespace))))
+      (key-value 12 "Created" ,time))))
+
+(kubernetes-ast-define-component ingress-line (state ingress)
+  (-let* ((current-time (kubernetes-state-current-time state))
+          (pending-deletion (kubernetes-state-ingress-pending-deletion state))
+          (marked-ingress (kubernetes-state-marked-ingress state))
+          ((&alist 'data data 'metadata (&alist 'name name 'creationTimestamp created-time)
+                   'spec (&alist 'rules ingress-rules)
+                   'status (&alist 'loadBalancer (&alist 'ingress ingress-lb-list)))
+           ingress)
+          (line `(line ,(concat
+                         ;; Name
+                         (format "%-45s " (kubernetes-utils-ellipsize name 45))
+
+                         ;; Hosts
+                         (format "%-25s " (--mapcat (alist-get 'host it) ingress-rules))
+
+                         ;; Address
+                         (format "%20s " (--mapcat (alist-get 'ip it) ingress-lb-list))
+
```

Comment:
Ports is missing.

Reply:
Can you attach the ingress json ? Will work with that.

Reply:
@ageekymonk I am afraid I cant because it has company information.

Reply:
Or may be I can mask it with random strings. Let me try.

Reply:
@ageekymonk I am finding it difficult to remove all the important information from the yaml. Maybe what you need is a longer host name